feature/metis/conversation/src/main/kotlin/de/tum/informatics/www1/artemis/native_app/feature/metis/conversation/ui/ConversationViewModel.kt

```diff
@@ -715,4 +727,87 @@ internal open class ConversationViewModel(
+            val fileName = resolveFileName(context, uri)
+            _selectedFileName.emit(fileName)
+            uploadFileOrImage(context)
+        }
+    }
+
+    private fun uploadFileOrImage(context: Context) {
+        viewModelScope.launch(coroutineContext) {
+            try {
+                val fileBytes = withContext(Dispatchers.IO) {
+                    _selectedFileUri.value?.let { uri ->
+                        context.contentResolver.openInputStream(uri)?.use { inputStream ->
+                            val fileSize = inputStream.available()
+
+                            val maxFileSize = 5 * 1024 * 1024
+                            if (fileSize > maxFileSize) {
+                                throw IllegalArgumentException(getString(context, R.string.conversation_vm_file_size_exceed))
+                            }
+                            inputStream.readBytes()
+                        }
+                    } ?: throw IllegalArgumentException(getString(context, R.string.conversation_vm_file_upload_failed))
+                }
+
+                val response = metisModificationService.uploadFileOrImage(
+                    context = metisContext,
+                    courseId = courseId,
+                    conversationId = conversationId,
+                    fileBytes = fileBytes,
+                    fileName = _selectedFileName.value,
+                    serverUrl = serverConfigurationService.serverUrl.first(),
+                    authToken = accountService.authToken.first()
+                )
+
+                val fileUploadResponse = (response as NetworkResponse.Response).data
+                val filePath = fileUploadResponse.path
+                    ?: throw IllegalArgumentException(getString(context, R.string.conversation_vm_file_upload_failed))
+
+                val currentText = newMessageText.value.text
+                val updatedText = if (isImage(_selectedFileName.value)) {
+                    "$currentText\n![${_selectedFileName.value}]($filePath)\n" // Image markdown
+                } else {
+                    "$currentText\n[${_selectedFileName.value}]($filePath)\n"  // File markdown
+                }
+                newMessageText.value = TextFieldValue(updatedText)
+                clearSelectedFile()
+            } catch (e: Exception) {
+                clearSelectedFile()
+                withContext(Dispatchers.Main) {
+                    Toast.makeText(
+                        context,
+                        e.message,
+                        Toast.LENGTH_SHORT
+                    ).show()
+                }
+            }
+        }
+    }
+
+    private fun clearSelectedFile() {
+        _selectedFileUri.value = null
+        _selectedFileName.value = ""
+    }
+
+    private fun resolveFileName(context: Context, uri: Uri?): String {
+        return try {
+            val resolver: ContentResolver = context.contentResolver
+            uri?.let {
```

Comment:
All the null handling (`?.let` ...) can be removed here aswell, as soon as all the `onFileSelected` types have been updated to a non-nullable `URI`